pkg/frontend/querymiddleware/astmapper/instant_splitting.go

```diff
@@ -34,13 +34,16 @@ var splittableVectorAggregators = map[parser.ItemType]bool{
 // Supported range vector aggregators
 
 const (
-	avgOverTime   = "avg_over_time"
-	countOverTime = "count_over_time"
-	increase      = "increase"
-	maxOverTime   = "max_over_time"
-	minOverTime   = "min_over_time"
-	rate          = "rate"
-	sumOverTime   = "sum_over_time"
+	avgOverTime     = "avg_over_time"
+	countOverTime   = "count_over_time"
+	increase        = "increase"
+	irate           = "irate"
+	maxOverTime     = "max_over_time"
+	minOverTime     = "min_over_time"
+	presentOverTime = "present_over_time"
+	rate            = "rate"
+	resets          = "resets"
```

Comment:
I'd avoid splitting `resets` because of this: https://github.com/grafana/mimir/pull/2592

While it could be a marginal issue for someone using `rate`, someone using `resets` _really_ wants that number, and we can't detect them all.

Reply:
Ah ok, that is a good catch 👀